Cleanup Redundant Code

drhodrid opened the pull request:
This branch cleans up several redundant aspects of the code that have not seen any development on master for many years (and, to my knowledge are no longer used). Thus far, I have removed support for: (i) reduced modelling (POD); (ii) spontaneous potentials (and associated test), which is a part of the porous media framework; (iii) the old solid-fluid coupling code and associated ALE mesh movement (and associated test). 

The changes are tested via buildbot, which is green: http://buildbot.ese.ic.ac.uk:8080/builders/cleanup_timeloop 

With that in mind, I am putting this up for a merge review now. This will allow anyone that is against these changes can speak up. Note that I am also keen to remove support for the following: (i) implicit solids; and (ii) porous media (thus cleaning up the code significantly) and I will do so unless there are any objections. **With that in mind, if there are any objections, PLEASE SPEAK UP NOW!!**

stephankramer commented on assemble/Free_Surface.F90:
```diff
@@ -2287,7 +2287,7 @@ subroutine move_mesh_free_surface(states, initialise, nonlinear_iteration)
 
     do i=1, size(states)
-      velocity => extract_vector_field(states(i), "Velocity")
+      velocity => extract_vector_field(states(i), "Velocity") ! Rhodri check with Steph
```
I think you didn't mean to include this in this purge

stephankramer commented on main/Fluids.F90:
```diff
@@ -550,14 +487,11 @@ SUBROUTINE FLUIDS()
           ewrite(1,*)'Start of another nonlinear iteration; ITS,nonlinear_iterations=',ITS,nonlinear_iterations
           ewrite(1,*)'###################'
 
+          ! Cope velocity to Iterated velocity for later convergence analysis:
```
*Copy* and this is not just about Velocity

stephankramer commented on main/Fluids.F90:
```diff
@@ -550,14 +487,11 @@ SUBROUTINE FLUIDS()
+          ! Calculate nonlinear velocity field:
           call relax_to_nonlinear(state)
-          call copy_from_stored_values(state, "Old")
+          call copy_from_stored_values(state, "Old")          
```
Trailing whitespace

drhodrid commented:
I have corrected the minor issues picked up by @stephankramer. Furthermore, I have removed support for porous media and implicit solids, in addition to legacy traffic options, as I mentioned previously. The manual has been updated to reflect these changes. Buildbot is green, so this is good for a final review. Accordingly, assigning to @jrper, given his expertise.

jrper commented:
As far as I can tell, (that's a lot of red text in the diff) this is only touching the things you claimed, and nothing which I know to be part of a live project, so I'm happy with the changes. I'd still recommend asking @matt-piggott for a final confirmation on that, since he's probably got an executive overview. We should also ask @tmbgreaves what the best way of tagging this point in the code history is, in case anyone needs to go in and recover any of this code (I'm thinking in particular of the various parasitic projects still using the Fluidity code base).

tmbgreaves commented:
James Percival wrote:
> @tmbgreaves what the best way of tagging

I think this is a reasonable candidate for a
minor-point release to be made, as we did for
the big PETSc change which forced use of a more
recent PETSc version.

Tim

drhodrid commented:
Hi Tim,

How would a minor point release work? Is this something you are able to manage post-merge, or does it need to be done beforehand?

Best wishes,

Rhod

On 23 Aug 2017, at 07:20, Tim Greaves <notifications@github.com<mailto:notifications@github.com>> wrote:

James Percival wrote:
> @tmbgreaves what the best way of tagging

I think this is a reasonable candidate for a
minor-point release to be made, as we did for
the big PETSc change which forced use of a more
recent PETSc version.

Tim

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/FluidityProject/fluidity/pull/165#issuecomment-324154943>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHXaKvEO8-aTre5yQJm6nFfKk4vUA29vks5sa0YBgaJpZM4O05Wf>.

tmbgreaves commented:
Pre-merge if possible, I think - I'll try to get it
done this week. Currently having a few technical
issues being away from my normal computers and having
lost a keyboard from a baby throwing up on it. On
screen keyboard on mobile phone not ideal for
detailed hacking but should have access to a better
systeṁ in the near future...

 

Rhodri Davies wrote:
> Hi Tim,
> 
> How would a minor point release work? Is this something you are able to manage
> post-merge, or does it need to be done beforehand?
> 
> Best wishes,
> 
> Rhod
> 
> On 23 Aug 2017, at 07:20, Tim Greaves <notifications@github.com
> <mailto:notifications@github.com>> wrote:
> 
> James Percival wrote:
> > @tmbgreaves what the best way of tagging
> 
> I think this is a reasonable candidate for a
> minor-point release to be made, as we did for
> the big PETSc change which forced use of a more
> recent PETSc version.
> 
> Tim
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<https://github.com/
> FluidityProject/fluidity/pull/165#issuecomment-324154943>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/
> AHXaKvEO8-aTre5yQJm6nFfKk4vUA29vks5sa0YBgaJpZM4O05Wf>.
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.*
>

tmbgreaves commented:
4.1.15 has been created, so as far as that's concerned you're clear to merge.

drhodrid commented:
OK great - thanks Tim. @matt-piggott -  waiting on you ;-)

jrper commented:
@drhodrid You might find email a faster method of communication than waiting for him to log on to github. Although I'd understood from @stephankramer that there was a request to save the implicit solids code.

drhodrid commented:
Thanks James. I'll email Matt. The request re: implicit solids has been dealt with offline and we've now got the go ahead to merge (thanks to the point release). I'll confirm with Matt and get this in...

Sent from my iPad

On 31 Aug 2017, at 16:09, James Percival <notifications@github.com<mailto:notifications@github.com>> wrote:


@drhodrid<https://github.com/drhodrid> You might find email a faster method of communication than waiting for him to log on to github. Although I'd understood from @stephankramer<https://github.com/stephankramer> that there was a request to save the implicit solids code.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/FluidityProject/fluidity/pull/165#issuecomment-326307078>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHXaKpIRY2nRPVBpT7D-cKT3CfFoAFCxks5sdr6ugaJpZM4O05Wf>.

drhodrid commented:
I've had the go ahead from @matt-piggott so will merge. Thanks @jrper and @stephankramer for comment/reviews.
